content/docs/formal-verification/_index.md

```diff
@@ -0,0 +1,137 @@
+modeled using two functions `encrypt` and `decrypt` satisfying the following equation:
+
+```js
+decrypt(key, encrypt(key, data)) = data
+```
+
+This allows the tool to replace any occurrence of the term `decrypt(key, encrypt(key, data))` with `data`, but it says
+nothing about the strength of the encryption algorithm. In a sense, it treats the encryption as perfect, since the only
+way to obtain the plaintext from the ciphertext is with knowledge of the correct key. Modeling cryptographic primitives
+in this way implies a number of trade-offs that may not be immediately obvious. On the one hand, it provides clean
+abstractions that allows us to specify the high-level building blocks of cryptographic protocols without having to worry
+too much about how each cryptographic primitive is implemented. This also allows us to instantiate the protocol with any
+primitive that satisfies the given equations and still be sure that the security proofs hold. On the other hand, it
+means that we cannot reason about some things like brute-force or padding-oracle attacks in a straight-forward manner.
+(If we want to do this, we need to model our primitives in a way that allow us to express these types of attacks within
+the model. Depending on the security properties we are interested in, this may either add complexity, which often has
+adverse effects on proving time, or may sometimes be impossible within the symbolic model.)
+
+Symbolic verification tools typically model the network as untrusted, allowing the attacker to intercept, delay, modify,
+and replay messages between participants at will. This is known as the _Dolev-Yao model_, and was first described in the
+paper [_On the Security of Public-Key Protocols_](https://www.cs.huji.ac.il/~dolev/pubs/dolev-yao-ieee-01056650.pdf).
+Individual tools often provide abstractions (like the `reliable-channel` builtin in Tamarin or the `passive` keyword in
+Verifpal) that can be used to restrict the capabilities of the attacker in different ways.
+
+Tools like [CryptoVerif](https://bblanche.gitlabpages.inria.fr/CryptoVerif/) analyze protocols in the _computational
+model_. Here, individual messages are modeled as bitstrings, cryptographic primitives are modeled as functions on
+bitstrings, and the adversary is modeled as an arbitrary probabilistic polynomial-time Turing Machine. The probability
+of the attacker successfully breaking the security of a given primitive has to be provided up-front by the user. The
+tool can then automatically prove indistinguishability between sequences of games (up to some probability), where the
+first game captures the analyzed protocol, and the final game is one where the required security property is obvious.
+The output is given as a bound on the probability of a successful attack as a function of the security parameters of the
+individual primitives used, and the number of protocol executions. A key benefit of the computational model is that this
```

Comment:
"Function of the advantages of the individual primitives" might work better here. In practice there's mainly a single security parameter for the entire protocol. AFAIK, by default Cryptoverif performs a concrete analysis rather than asympotic one, which would also justify omitting security parameters from the discussion.